include/grpcpp/impl/codegen/server_context_impl.h

```diff
@@ -272,6 +279,22 @@ class ServerContext {
   /// Applications never need to call this method.
   grpc_call* c_call() { return call_; }
 
+  /// NOTE: This is an API for advanced users who need custom allocators.
+  /// Get and maybe mutate the allocator state associated with the current RPC.
+  /// Currently only applicable for callback unary RPC methods.
+  /// WARNING: This is experimental API and could be changed or removed.
+  ::grpc::experimental::RpcAllocatorState* GetRpcAllocatorState() {
+    return message_allocator_state_;
+  }
+
+  /// Get the default unary reactor for use in minimal reaction cases.
+  /// WARNING: This is experimental API and could be changed or removed.
+  experimental::ServerUnaryReactor* DefaultReactor() {
```

Comment:
I'd like to ban it so that we can do something in this case later (like lazy initialization as Alan suggested).

Reply:
But I don't know if banning it is the right decision.

Reply:
So at the current time, it's ok to call it multiply. Maybe we can decide if it should be that way.

Reply:
What should a user expect? If a user calls this twice, are they going to think that they got two fresh reactors? Are they going to think they need to free each one, especially since the typical non-DefaultReactor() use case has users deeply allocating the reactors? I'm concerned that users will get very confused by this.

Alternatively, we could provide a factory method that constructs a default reactor.

Reply:
Added comments to make the calling convention more clear and explicitly forbid multiple calls.